content/en/blogs/transform-traditional-applications-into-microservices.md

```diff
@@ -0,0 +1,227 @@
+
+4. The deployment template must contain annotations (required during automatic sidecar injection of Istio).
+
+```bash
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "true"
+```
+
+5. The service and deployment contain annotations. The KubeSphere CRD Controller automatically matches VirtualService and DestinationRules to the service.
+
+```bash
+# Service
+kind: Service
+metadata:
+  annotations:
+    servicemesh.kubesphere.io/enabled: "true"
+
+# Deployment
+kind: Deployment
+metadata:
+  annotations:
+    servicemesh.kubesphere.io/enabled: "true"
+```
+
+## Example
+
+To implement traffic monitoring, two independent applications are required (for example, WordPress and MySQL). After the two applications work properly, we then refactor them into KubeSphere microservices and inject the sidecar. 
+
+Open the [MySQL Docker Hub](https://hub.docker.com/_/mysql "mysql dockerhub") page, and you can see explanation of the `MYSQL_ROOT_PASSWORD` variable. On the KubeSphere web console, set the default MySQL password.
+
+Open the [WordPress Docker Hub](https://hub.docker.com/_/wordpress "wordpress dockerhub") page, and you can see three database variables `WORDPRESS_DB_PASSWORD` `WORDPRESS_DB_USER` `WORDPRESS_DB_HOST`. On the KubeSphere web console, set values of the three variables to connect Wordpress to MySQL.
+
+## Create a Traditional Application
+
+First, create a workspace and a project with the gateway and tracing features enabled.
+
+![00-enable-gateway](/images/blogs/transform-traditional-applications-into-microservices/00-enable-gateway.png)
+
+Select **Application Workloads** > **Service**. On the **Service** page, click **Create**. On the **Create Service** page, click **Stateful Service** to create a MySQL service.
+
+![01-create-mysql](/images/blogs/transform-traditional-applications-into-microservices/01-create-mysql.png)
+
+![02-create-mysql](/images/blogs/transform-traditional-applications-into-microservices/02-create-mysql.png)
+
+In **Environment Variables**, set the default password.
+
+![03-set-password](/images/blogs/transform-traditional-applications-into-microservices/03-set-password.png)
+
+Likewise, create a stateless WordPress service. 
+
+![04-create-wp](/images/blogs/transform-traditional-applications-into-microservices/04-create-wp.png)
+
+![05-create-wp](/images/blogs/transform-traditional-applications-into-microservices/05-create-wp.png)
+
+The following variables are for demonstration only. In production, select **Use ConfigMap or Secret**.
+
+![06-create-wp](/images/blogs/transform-traditional-applications-into-microservices/06-create-wp.png)
+
+Select **Edit External Access**, and then change the access mode to **NodePort**.
+
+![07-nodeport](/images/blogs/transform-traditional-applications-into-microservices/07-nodeport.png)
+
+After the pods run properly, access the service page using the service IP address + node port. It can be seen that the application runs properly.
```

Comment:
`using the service IP address + node port`
->
`at <Service IP address>:<Node port>`